CMakeLists.txt

```diff
@@ -11,6 +11,10 @@ set( CMAKE_EXPORT_COMPILE_COMMANDS ON )
 set( CMAKE_CXX_STANDARD 17 )
 
 # dependencies
+if ( MSVC )
+    # default prebuilt boost installation directory
+    set( BOOST_ROOT "c:/local/boost_1_70_0" )
```

Comment:
What happens if the user supplies -DBOOST_ROOT on the CMake command line? Will this override it?

Reply:
Good catch. I've changed it so that it only provides a default cached value - the user's selection will override it, and will persist across cmake invocations in a given build folder unless the cmake cache is deleted.

Reply:
I checked and unfortunately this does prevent the user from supplying BOOST_ROOT on the CMake command line. But this works:
```
if( NOT BOOST_ROOT )
    set( BOOST_ROOT "c:/local/boost_1_70_0" )
endif()
```
Would you be willing to change it? I like the PR otherwise. Thanks.

Reply:
Sorry, bad timing :) I will read the new code.

Reply:
Sure I can change it. You did check in a fresh build folder, right?